### ISSUE TITLE
PyBEL API test failing

### ISSUE THREAD

**Issue by bgyori** [sorgerlab/indra#369]
One of the tests has been erroring on Travis (PyBEL is cloned from github in .travis.yml). The same test passes with the latest version of PyBEL installed from PyPI.
```
ERROR: test_pybel_api.test_phosphorylation_one_site_with_evidence
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/bgyori/indra/indra/tests/test_pybel_api.py", line 298, in test_phosphorylation_one_site_with_evidence
    pbp = bel.process_pybel_graph(g)
  File "/home/travis/build/bgyori/indra/indra/sources/bel/bel_api.py", line 127, in process_pybel_graph
    bp.get_statements()
  File "/home/travis/build/bgyori/indra/indra/sources/bel/pybel_processor.py", line 101, in get_statements
    self._get_modification(u_data, v_data, d)
  File "/home/travis/build/bgyori/indra/indra/sources/bel/pybel_processor.py", line 201, in _get_modification
    ev = _get_evidence(u_data, v_data, edge_data)
  File "/home/travis/build/bgyori/indra/indra/sources/bel/pybel_processor.py", line 456, in _get_evidence
    epistemics['section_type'] = _pybel_text_location_map.get(text_location)
TypeError: unhashable type: 'dict'
```

**Comment by johnbachman** [comment]
@cthoyt, do you have any thoughts on this? The test passes locally for me with the latest pybel/master from Github, but for some reason we are still getting this error on Travis. What appears to be happening is that `text_location`, which is pulled out an edge annotation created by `add_qualified_edge` in the test shown above, is retrieving a dict instead of a string; this causes the call to `get` to give the TypeError.

**Comment by johnbachman** [comment]
@cthoyt As it turns out the latest master 0.10.1, commit f5c1a9a82c0bf8fe7 returns a string as expected, whereas it looks like the branch develop, commit ff7bdfbd107dcab1d6f, returns a dict. It looks like  `pip install git+https://github.com/pybel/pybel.git` is getting the develop version.

**Comment by cthoyt** [comment]
Muahaha! I am an evil developer who pays no attention to his users. I've updated the data modal within the BELGraph class so edges' `ANNOTATION` entries are now always dictionaries, instead of strings. Now, when BELGraphs are parsed, a relation with a list of annotations will not get exploded into multiple edges, but it can actually all stay together in one nice package. I was planning on submitting a PR to INDRA to fix this once I made a release, but you guys are just too on top of things!

Since for the most part INDRA is ignoring annotations, I hope this doesn't have any other negative side-effects

**Comment by bgyori** [comment]
Fixed in #377

**Comment by bgyori** [comment]
@cthoyt just merged develop into master so now the error cropped up again and we actually need to deal with it.

**Comment by bgyori** [comment]
I added a basic fix in 417a7576dc0e29237bafd15cd0c25b66fd4c71f4
